src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/X509SecurityTokenProvider.cs

```diff
@@ -34,9 +34,9 @@ internal X509SecurityTokenProvider(X509Certificate2 certificate, bool clone)
             }
         }
 
-        protected override async Task<SecurityToken> GetTokenCoreAsync(CancellationToken cancellationToken)
+        protected override Task<SecurityToken> GetTokenCoreAsync(CancellationToken cancellationToken)
         {
-            return await Task.FromResult<SecurityToken>(new X509SecurityToken(certificate: _certificate, clone: _clone, disposable:_clone));
+            return Task.FromResult<SecurityToken>(new X509SecurityToken(certificate: _certificate, clone: _clone, disposable:_clone));
```

Comment:
We don't need to specify the generic type on FromResult as the compiler can work it out. I know this was already there, but as this line is getting changed, it would be worth doing this little bit of cleanup too.